mathesar_ui/src/stores/table-data/constraints.ts

```diff
@@ -63,11 +63,17 @@ export class ConstraintsDataStore implements Writable<ConstraintsData> {
     return getStoreValue(this.store);
   }
 
-  async fetch(): Promise<ConstraintsData> {
-    this.update((existingData) => ({
-      ...existingData,
-      state: States.Loading,
-    }));
+  async fetch({
+    showLoading = true,
+  }: {
+    showLoading?: boolean,
+  } = {}): Promise<ConstraintsData> {
+    if (showLoading) {
+      this.update((existingData) => ({
+        ...existingData,
+        state: States.Loading,
+      }));
```

Comment:
The state property in each store is meant to only reflect the state of the store itself. As of now, the data layer (stores) is completely separate from the view layer (components), and only the view depends on the data, and not the other way around.

Setting the state based on a conditional passed down from the component ties it to the view layer. It would be better to handle this on the component, instead of the store.

In the component, we can show the spinner only if constraints is empty and state is loading.

Reply:
Addressed in 087452ba0941b2690a0d20bc81d5025c0713a65c